frame/nomination-pools/src/lib.rs

```diff
@@ -1599,6 +1620,59 @@ pub mod pallet {
 			Ok(())
 		}
 
+		// TODO: Update Weight info
+		#[pallet::weight(
+			T::WeightInfo::bond_extra_transfer()
+			.max(T::WeightInfo::bond_extra_reward())
+		)]
+		pub fn root_bond_extra(
+			origin: OriginFor<T>, 
+			member_account: AccountIdLookupOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let member_account = T::Lookup::lookup(member_account)?;
+			let (mut member, mut bonded_pool, mut reward_pool) =
+				Self::get_member_with_pools(&member_account)?;
+
+			ensure!(bonded_pool.is_root(&who), Error::<T>::NotRoot);
+			ensure!(ClaimableAction::<T>::get(&member_account) == BondExtraSource::Operator, Error::<T>::DoesNotHavePermission);
+
+			// IMPORTANT: reward pool records must be updated with the old points. why?
+			reward_pool.update_records(bonded_pool.id, bonded_pool.points)?;
```

Comment:
Should reuse code rather than duplicating.

Reply:
Looks like this is no longer outstanding.